### DISCUSSION TITLE
JavaScript Executor with Arguments

### DISCUSSION THREAD

**Discussion by vishallanke** [cypress-io/cypress#19046] [Questions and Help]
### Current behavior

I want to migrate below code(Protractor) in Cypress. But it seems there is no way. We tried with few combinations

```
async draw(option: DrawOption): Promise<void> {
return browser.executeScript('return window.draw(arguments[0])', option);
}
```

options is Enum

Below code works. But there is no way to send options enum as an argument
```
cy.window().then((win) => {
 win.eval('window.draw(arguments[0])');
});

```

### Desired behavior

Able to pass arguments to JavaScript executor

### Test code to reproduce

Code from Protector
```
async draw(option: DrawOption): Promise<void> {
return browser.executeScript('return window.draw(arguments[0])', option);
}
```

### Cypress Version

8.5

**Comment by bahmutov** [comment]
@vishallanke it is much simpler, read https://glebbahmutov.com/cypress-examples/commands/window.html

**Comment by vishallanke** [comment]
@bahmutov  Thanks for links. Team went through this and tried few things. But, JS with arguments does not work out. Please check below example,

Working Protector code
```
async draw(option: DrawOption): Promise<void> {
return browser.executeScript('return window.draw(arguments[0])');
}
```

Working Cypress code for above protector code
```
cy.window().then((win) => {
 win.eval('window.draw(arguments[0])');
});
```

So, no issues in above code.

But below code does not work when I pass Enum model to it
Working Protector code where options is Model data of type Enum
```
async draw(option: DrawOption): Promise<void> {
return browser.executeScript('return window.draw(arguments[0])', options);
}
```

Cypress code for above protector code. Its not working. Here we tried many things.
```
cy.window().then((win) => {
 win.eval('window.draw(arguments[0])', options);
});
```

```
cy.window().then((win) => {
 win.eval('window.draw(arguments[0]), options');
});
```

**Comment by bahmutov** [comment]
You are overthinking this. There is no need to use `eval` at all. Just call the method on the yielded `win`

```js
cy.window().then((win) => {
 win.draw(... you arguments);
});
```

If you are not sure, put the `debugger` keyword there and see for yourself in DevTools

```js
cy.window().then((win) => {
 debugger
 win.draw(... you arguments);
});
```

**Comment by chrisbreiding** [comment]
Issues in our GitHub repo are reserved for potential bugs or feature requests. This issue will be closed and converted into a [discussion](https://github.com/cypress-io/cypress/discussions) since it appears to be neither a bug nor a feature request.

We recommend questions relating to how to use Cypress be asked in [GitHub Discussions](https://github.com/cypress-io/cypress/discussions) or our [community chat](https://gitter.im/cypress-io/cypress). Also try searching our [existing GitHub issues](https://gitter.im/cypress-io/cypress/issues), reading through our [documentation](https://docs.cypress.io), or searching [Stack Overflow](https://stackoverflow.com/questions/tagged/cypress) for relevant answers.
